src/Service/Resolvers/Sql Query Structures/SqlQueryStructure.cs

```diff
@@ -395,19 +395,19 @@ private List<OrderByColumn> PrimaryKeyAsOrderByColumns()
                     AddColumnsForEndCursor();
                 }
 
-                // if the user does a paginated query only requesting hasNextPage
-                // there will be no elements in Columns
-                if (!Columns.Any())
-                {
-                    AddColumn(PrimaryKey()[0]);
-                }
-
                 if (PaginationMetadata.RequestedHasNextPage)
                 {
                     _limit++;
                 }
             }
 
+            // If there are no columns, add the primary key column
+            // to prevent failures when executing the database query.
+            if (!Columns.Any())
+            {
+                AddColumn(PrimaryKey()[0]);
```

Comment:
what if there are multiple primary key columns? Are we choosing only the first? 

For pagination, it may make sense to still hit the database by creating a query so that it gets the hasNextPage field correctly.
But for the cases when there is no database column to project or no dependency on database results why do we even try to hit the database?

Reply:
Yes, when the PK is composite, only the first key/column is chosen. Having just one valid column in the `select` query makes it run successfully. I have added a test case for this scenario where the Pk is composite. 


There are only very few cases where it is possible to completely avoid hitting the database. 
Only in this case (and similar one for stored procedure),
```graphql
{
  books{
    __typename
  }
}
```
it might be possible to avoid hitting the database. 


In all other cases, such as these, the result depends on the items present in the database, policies defined etc.

```graphql

  {
       books{
         items{
           __typename
         }
       }
   }

```
- In the above query with `filter` or `first` defined would necessitate querying the database.

- Incase of nested queries, where some of the selection sets contain entity fields while the rest contain only `__typename` such as this
 
```graphql
{
  publishers {
    items {
      id
      books {
        items {
          __typename
        }
      }
    }
  }
}
```
also necessitates that we query the database.